### ISSUE TITLE
Filter wc_myparcel_order_delivery_options no longer working as expected

### ISSUE THREAD

**Issue by DarkAllMan** [myparcelnl/woocommerce#844]
### Plugin version

4.12.2

### WooCommerce version

6.6.1

### WordPress version

6.0

### PHP version

8.0.20

### What went wrong?

In order for me to get Table Rates working with MyParcel, I use the filter wc_myparcel_order_delivery_options.
Since the last update, the changes I push with this filter, are not stored in the order.

For instance, when I return the following object:
```
WCMP_DeliveryOptionsFromOrderAdapter Object
(
    [carrier:protected] => 
    [date:protected] => 
    [deliveryType:protected] => standard
    [packageType:protected] => digital_stamp
    [pickupLocation:protected] => 
    [shipmentOptions:protected] => WCMP_ShipmentOptionsFromOrderAdapter Object
        (
            [signature:protected] => 
            [only_recipient:protected] => 
            [age_check:protected] => 
            [same_day_delivery:protected] => 
            [large_format:protected] => 
            [return:protected] => 
            [insurance:protected] => 
            [label_description:protected] => 
        )

)
```

The order is still saved as a package and not as a digital stamp.

### Reproduction steps

```
	add_filter( 'wc_myparcel_order_delivery_options', array ( $this, 'set_Myparcel_Delivery_Options' ), 10, 2 );

	public function set_Myparcel_Delivery_Options( $deliveryOptions, $order ){	
		
		if(is_admin()){
			return $deliveryOptions;
		}
		
		$data = [
			'package_type' => $this->getCurrentPackageTypeFromOrder($order)
		];
		$deliveryOptions = new WCMP_DeliveryOptionsFromOrderAdapter($deliveryOptions, $data);
		
		return $deliveryOptions;
	}
```

### Additional context

Before filter:
```
WCMP_DeliveryOptionsFromOrderAdapter Object
(
    [carrier:protected] => 
    [date:protected] => 
    [deliveryType:protected] => standard
    [packageType:protected] => 
    [pickupLocation:protected] => 
    [shipmentOptions:protected] => WCMP_ShipmentOptionsFromOrderAdapter Object
        (
            [signature:protected] => 
            [only_recipient:protected] => 
            [age_check:protected] => 
            [same_day_delivery:protected] => 
            [large_format:protected] => 
            [return:protected] => 
            [insurance:protected] => 
            [label_description:protected] => 
        )

)
```
after filter:

```
WCMP_DeliveryOptionsFromOrderAdapter Object
(
    [carrier:protected] => 
    [date:protected] => 
    [deliveryType:protected] => standard
    [packageType:protected] => digital_stamp
    [pickupLocation:protected] => 
    [shipmentOptions:protected] => WCMP_ShipmentOptionsFromOrderAdapter Object
        (
            [signature:protected] => 
            [only_recipient:protected] => 
            [age_check:protected] => 
            [same_day_delivery:protected] => 
            [large_format:protected] => 
            [return:protected] => 
            [insurance:protected] => 
            [label_description:protected] => 
        )

)
```

**Comment by wthijmen** [comment]
Hi @DarkAllMan,

I did research on your issue. I can not find any changes with the current version.
I can **highly recommend** you to try it out on an **older version** of the plugin to make sure in which version your code is not working anymore.

Is it possible for you to send us some more information especially about the following function:
`$this->getCurrentPackageTypeFromOrder($order)`

You can share all the information by sending it to [support@myparcel.nl](mailto:support@myparcel.nl).

**Comment by DarkAllMan** [comment]
Hi @wthijmen,

In my live site I rolled back to version:  4.11.0
There it still works.

I will share my solution for BETRS today via support@myparcel.nl

**Comment by DarkAllMan** [comment]
@wthijmen Did you receive my email as I did not get any confirmation?

**Comment by DarkAllMan** [comment]
@wthijmen Can someone please confirm my email was received and the issue is being investigated?

**Comment by DarkAllMan** [comment]
I had a nice session with @Barry020 today. I showed him the issue and he is investigating.

**Comment by RichardPerdaan** [comment]
fixed in version [v4.13.0](https://github.com/myparcelnl/woocommerce/releases/tag/v4.13.0)
